docs/license-handling.md

```diff
@@ -0,0 +1,31 @@
+# License handling
+
+ORT deals with various types of licenses. Their ORT-specific names and purposes are explained in detail below.
+
+## Declared license
+
+For ORT, declared licenses are those licenses which are declared as part of package manager metadata, and as gathered
+by the ORT *analyzer*. In other words, this is the license the author of the package claims or intends the package to
+be licensed under; the license which is "visible from the outside". However, the declared license alone only provides an
+incomplete picture, e.g. in so-called "envelope cases" where the license visible from the outside on the envelope does
+not match what is inside the envelope, i.e. in the source code. So, license information e.g. inside a `LICENSE` file is
+*not* a declared license in ORT-sense.
```

Comment:
The last sentence needs to be improved to make clearer how this is related to the "envelope cases".
You could add 1 or 2 examples of where we get declared licenses from, e.g. links to Maven and NPM docs of the respective fields.